prime-router/src/main/kotlin/fhirengine/engine/CustomFhirPathFunctions.kt

```diff
@@ -145,4 +164,140 @@ class CustomFhirPathFunctions : FhirPathFunctions {
+                metadata
+            )
+            GeoData.DataTypes.POSTAL_CODE -> getRandomGeoValue(
+                parameters[1].first().primitiveValue(),
+                GeoData.ColumnNames.ZIP_CODE,
+                metadata
+            )
+            GeoData.DataTypes.TESTING_LAB -> "Any lab USA"
+            GeoData.DataTypes.SENDER_IDENTIFIER -> UUID.randomUUID().toString()
+            GeoData.DataTypes.FACILITY_NAME -> "Any facility USA"
+            GeoData.DataTypes.NAME_OF_SCHOOL -> "Any Fake School"
+            GeoData.DataTypes.REFERENCE_RANGE -> randomChoice("", "Normal", "Abnormal", "Negative")
+            GeoData.DataTypes.RESULT_FORMAT -> "CWE"
+            GeoData.DataTypes.PATIENT_PREFERRED_LANGUAGE -> randomChoice("ENG", "FRE", "SPA", "CHI", "KOR")
+            GeoData.DataTypes.PATIENT_COUNTRY -> "USA"
+            GeoData.DataTypes.SITE_OF_CARE -> FakeReport.getRandomSiteOfCare()
+            GeoData.DataTypes.PATIENT_AGE_AND_UNITS -> getPatientAgeAndUnits()
+            GeoData.DataTypes.COUNTY -> getRandomGeoValue(
+                parameters[1].first().primitiveValue(),
+                GeoData.ColumnNames.COUNTY,
+                metadata
+            )
+            GeoData.DataTypes.EQUIPMENT_MODEL_NAME -> randomChoice(
+                    "LumiraDx SARS-CoV-2 Ag Test",
+                    "BD Veritor System for Rapid Detection of SARS-CoV-2"
+                )
+            GeoData.DataTypes.TEST_PERFORMED_CODE -> randomChoice(
+                "95209-3",
+                "94558-4"
+            )
+            GeoData.DataTypes.OTHER_TEXT -> "I am some random text"
+            GeoData.DataTypes.BLANK -> ""
+            GeoData.DataTypes.TEXT_OR_BLANK -> randomChoice("I am some random text", "")
+            GeoData.DataTypes.NUMBER -> Random.nextInt().toString()
+            GeoData.DataTypes.DATE -> DateUtilities.getDateAsFormattedString(
+                getRandomDate().toInstant(),
+                DateUtilities.datePattern
+            )
+            GeoData.DataTypes.BIRTHDAY -> DateUtilities.getDateAsFormattedString(
+                getRandomDate().toInstant(),
+                DateUtilities.datePattern
+            )
+            GeoData.DataTypes.DATETIME -> DateUtilities.getDateAsFormattedString(
+                getRandomDate().toInstant(),
+                DateUtilities.datetimePattern
+            )
+            GeoData.DataTypes.HD -> "0.0.0.0.1"
+            GeoData.DataTypes.EI -> "SomeEntityID"
+            GeoData.DataTypes.ID -> Faker().numerify("######")
+            GeoData.DataTypes.ID_CLIA -> Faker().numerify("##D#######")
+            GeoData.DataTypes.ID_DLN -> Faker().idNumber().valid()
+            GeoData.DataTypes.ID_SSN -> Faker().idNumber().invalidSvSeSsn()
+            GeoData.DataTypes.ID_NPI -> NPIUtilities.generateRandomNPI(Faker())
+            GeoData.DataTypes.STREET -> Faker().address().streetAddress()
+            GeoData.DataTypes.PERSON_NAME -> randomChoice(
```

Comment:
Is there a reason why we can't use `Faker().name().fullname` here?
It looks like you can even get a random Harry Potter character `Faker().harryPotter().character()`